src/controllers/post.js

```diff
@@ -40,3 +40,30 @@ export const getById = async (req, res) => {
     return sendDataResponse(res, 500, 'Unable to get post')
   }
 }
+
+export const updatePost = async (req, res) => {
+  const id = parseInt(req.params.id)
+  const { content } = req.body
+
+  if (!id) {
+    return sendDataResponse(res, 400, { content: 'Must provide id' })
+  }
+
+  if (!content) {
+    return sendDataResponse(res, 400, { content: 'Must provide content' })
+  }
+
+  const checkIfPostExists = await Post.findById(id)
+  if (!checkIfPostExists) {
+    return sendDataResponse(res, 404, { id: 'Post not found' })
+  }
+
```

Comment:
```JavaScript
const { user } = req // Logged in user, which has been added to the request object by the middleware form JWT

  if (user.id !== checkIfPostExists.userId) {
    return sendDataResponse(res, 403, {
      message: 'You are only allowed to update your own posts'
    })
  }

I can do this. Seems to be working after testing.
```

Reply:
Can you push this code to the new branch so i can test it